napari/_vispy/overlays/base.py

```diff
@@ -6,17 +6,25 @@
 
 
 class VispyBaseOverlay:
-    def __init__(self, *, overlay, node, parent):
+    """
+    Base overlay backend for vispy.
+
+    Creates event connections between napari Overlay models and the
+    vispy backend, translating them into rendering.
+    """
+
+    def __init__(self, *, overlay, node, parent=None):
         super().__init__()
         self.overlay = overlay
 
         self.node = node
-        self.node.parent = parent
         self.node.order = self.overlay.order
 
         self.overlay.events.visible.connect(self._on_visible_change)
         self.overlay.events.opacity.connect(self._on_opacity_change)
 
+        self.node.parent = parent
```

Comment:
Would we like to reset it here even if the node has parents?

Reply:
I'm not sure what you're asking, but I didn't put `reset()` here for the same reason why `VispyBaseLayer` doesn't have it: the subclasses need to have a `reset()` cann *at the end* of all the setup, otherwise things break.

Reply:
This statement overwrites `parent` of `node`. But also `parent` is an optional argument of the constructor. So it may happen that node has a parent, someone calls the constructor, without providing a parent and it will set `node.parent` to `None`.

Reply:
Ah I see... Well, I think we assume in every case here that, during construction, node is not yet parented. If you see the subclasses, they all create the node during construction, and are supposed to not do anything special with parenting. Does that work for you?

Reply:
```suggestion
        if parent is not None:
            self.node.parent = parent
```
So maybe this? Or making parent mandatory?

Reply:
I guess this works!